### PR TITLE
[feat]: [PIE-4327]: Integration of RepoURL and file path verification on pipeline and input set import flow #35476  Closed

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes
Integration of RepoURL and file path verification on pipeline and input set import flow. If the file is already being used on git, we need to throw a warning or continue with the entity save, based on the user request. This will require a flag in the API to see whether we have to warn, or proceed.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java [hunk 1]
```diff
@@ -434,4 +440,32 @@ public static void checkAndThrowMismatchInImportedPipelineMetadata(String orgIde
           invalidFields);
     }
   }
+
+  public PipelineEntity handleImportFlow(String accountId, String orgIdentifier, String projectIdentifier, String importedPipelineYAML) {
+    GitEntityInfo gitEntityInfo = GitAwareContextHelper.getGitRequestParamsInfo();
+    Scope scope = Scope.of(accountId, orgIdentifier, projectIdentifier);
+    String repoURL =
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java:447]: Can you create a method in GitAwareEntityHelper that calls scmGitSyncHelper to get the repo URL? We should not call scmGitSyncHelper from our service layer

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java [hunk 2]
```diff
@@ -432,11 +435,12 @@ public PipelineEntity importPipelineFromRemote(String accountId, String orgIdent
     String importedPipelineYAML =
         pmsPipelineServiceHelper.importPipelineFromRemote(accountId, orgIdentifier, projectIdentifier);
 
-    PMSPipelineServiceHelper.checkAndThrowMismatchInImportedPipelineMetadata(
-        orgIdentifier, projectIdentifier, pipelineIdentifier, pipelineImportRequest, importedPipelineYAML);
+    PMSPipelineServiceHelper.checkAndThrowMismatchInImportedPipelineMetadata(orgIdentifier, projectIdentifier,
+        pipelineIdentifier, pipelineImportRequest, importedPipelineYAML);
+
+    PipelineEntity pipelineEntity = pmsPipelineServiceHelper.handleImportFlow(
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java:441]: this should be the first thing you do, before even importing the pipeline from git

**Review by namanvermah — CHANGES_REQUESTED** [review]

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 18 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 447 at r2](https://reviewable.io/reviews/harness/harness-core/35573#-N7URGvfncZjic-GwopW-r2-447:-N7VIGgg2eFNo6B3Qfty:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2aa438506bcab8525ea7e85267d49829d498ecc6/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L447)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

Can you create a method in GitAwareEntityHelper that calls scmGitSyncHelper to get the repo URL? We should not call scmGitSyncHelper from our service layer
</blockquote></details>

Done.
___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java` line 441 at r2](https://reviewable.io/reviews/harness/harness-core/35573#-N7URGvfncZjic-GwopX-r2-441:-N7VII-T8OZ6iQ-hU6Fl:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2aa438506bcab8525ea7e85267d49829d498ecc6/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java#L441)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

this should be the first thing you do, before even importing the pipeline from git
</blockquote></details>

Done.

**Review by senjucanon2 — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 19 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
None of these changes made in the files are related to delegate tasks, hence the changes are backward compatible

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 13 of 15 files at r2, 5 of 5 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573#-:-N7VRxSm9WSvJcfQtjuH:bnfp4nl)* status: 20 file(s) to review. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 452 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VWwcuE29oDI70FLdT:-N7VWwcuE29oDI70FLdU:b-s8mnjg) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L452)):*
> ```Java
>         throw new GitYamlException("The Requested YAML has already been imported.");
>       } else {
>         log.info("Importing YAML forcefully with Pipeline Id : {}", pipelineIdentifier);
> ```

we should add repo url and filepath as well in the log coz thats the uniqueness that we are breaking at this point.

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VXdmf0_x0pa9yJSDd:-N7VXdmf0_x0pa9yJSDe:b-v9umyi) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
> ```Java
> 
>     if (isAlreadyImported(totalInstancesOfYAML)) {
>       if (!gitEntityInfo.isForceImport()) {
> ```

if its force import, then we actually don't need to do the query in DB. We can simply skip it.

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java` line 194 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VYAI_AuW34sS2mkPY:-N7VYAI_AuW34sS2mkPZ:b-op3hrm) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java#L194)):*
> ```Java
>   }
> 
>   public String getRepoUrl(String accountId, String orgIdentifier, String projectIdentifier) {
> ```

it should be "accountIdentifier"

**Comment by mohitg0795** [comment]
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java` line 70 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VYLIM76QkIm8ktpQu:-N7VYLIM76QkIm8ktpQv:b-bi708h) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java#L70)):*
> ```Java
>     final String lastCommitId =
>         getRequestParamFromContext(GitSyncApiConstants.LAST_COMMIT_ID, pathParameters, queryParameters);
>     final String isForceImport =
> ```

This isn't a git property, then why is it included in git sync thread decorator? We should add it as flag in request object itself.

**Review by namanvermah — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 20 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java` line 70 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VYLIM76QkIm8ktpQu:-N7ZX-l093e_syRGrWx3:b4xmuez) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java#L70)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

This isn't a git property, then why is it included in git sync thread decorator? We should add it as flag in request object itself.
</blockquote></details>

because it makes sense for it to be in the thread decorator. Its high time we rethink what the sdk is supposed to be because there's way to much of common logic being asked shoved out of it.

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 3 of 4 files at r6, 6 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573#-:-N7njDNV1aBz811lOylm:bnfp4nl)* status: 24 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 15 files at r2, 1 of 5 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 3 of 4 files at r6, 5 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 12 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 452 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VWwcuE29oDI70FLdT:-N7o1p_H3W61COK1rk1k:b32v57) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L452)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

we should add repo url and filepath as well in the log coz thats the uniqueness that we are breaking at this point.
</blockquote></details>

+1
___
*[`980-commons/src/main/java/io/harness/exception/GitYamlException.java` line 7 at r7](https://reviewable.io/reviews/harness/harness-core/35573#-N7o29Gw51Wfd4OpvaUc:-N7o29Gw51Wfd4OpvaUd:b-xhp3vr) ([raw file](https://github.com/harness/harness-core/blob/efd43ce36e4dd24f77862ed8454674afb3b6125f/980-commons/src/main/java/io/harness/exception/GitYamlException.java#L7)):*
> ```Java
>     super(message);
>   }
> }
> ```

OwnedBy is missing



_Code quote:_
```Java
public class GitYamlException extends InvalidRequestException {
  public GitYamlException(String message) {
    super(message);
  }
}
```

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 12 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VXdmf0_x0pa9yJSDd:-N7o66WkAdR2wFcshmUT:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

if its force import, then we actually don't need to do the query in DB. We can simply skip it.
</blockquote></details>

Done.
___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 452 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VWwcuE29oDI70FLdT:-N7o6C5KAY2GEwDn-RCA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L452)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java` line 194 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VYAI_AuW34sS2mkPY:-N7o6K-o626EiI6DE2uX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java#L194)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

it should be "accountIdentifier"
</blockquote></details>

Done.
___
*[`980-commons/src/main/java/io/harness/exception/GitYamlException.java` line 7 at r7](https://reviewable.io/reviews/harness/harness-core/35573#-N7o29Gw51Wfd4OpvaUc:-N7o6LxJ8xCgQu-6odpk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/efd43ce36e4dd24f77862ed8454674afb3b6125f/980-commons/src/main/java/io/harness/exception/GitYamlException.java#L7)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

OwnedBy is missing
</blockquote></details>

Done.
___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java` line 70 at r5](https://reviewable.io/reviews/harness/harness-core/35573#-N7VYLIM76QkIm8ktpQu:-N7o6N1i2eYCXXcTgPpC:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d5d0f2f541799acf52d802c44584ee4ece7e4489/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitSyncThreadDecorator.java#L70)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

No worries, we can have a discussion on it. Please call out a meeting with concerned stakeholders. We can discuss and close it out.

IMO, we need to move away from git sync thread decorator as much as possible. ForceImport is totally a entity level operation flag that we are using, so it should flow via usual arguments route rather than a thread route.
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 455 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pBTr3CMzc7ZyeztTw:-N7pBTr3CMzc7ZyeztTx:b7ehb6l) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L455)):*
> ```Java
>   }
> 
>   public boolean isAlreadyImported(String accountId, String repoURL, String filePath) {
> ```

this shouldn't be public. If required in UTs, lets make it protected.



_Code quote:_
```Java
public 
```

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 455 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pBfaSC5SYSIXbX1FM:-N7pBfaSC5SYSIXbX1FN:b-cpupr) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L455)):*
> ```Java
>   }
> 
>   public boolean isAlreadyImported(String accountId, String repoURL, String filePath) {
> ```

should be accountIdentifier

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 440 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pChtw9PQZasdYmCBI:-N7pChtw9PQZasdYmCBJ:b-gdi8sn) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L440)):*
> ```Java
>   }
> 
>   public String getRepoUrlAndCheckForFileUniqueness(String accountId, String orgIdentifier, String projectIdentifier,
> ```

accountIdentifier*

**Comment by mohitg0795** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pD4oGCynfJs5kBEOJ:-N7pD4oGCynfJs5kBEOK:b-88yg0v) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
> ```Java
>           gitEntityInfo.getFilePath());
>     }
>     if (!isForceImport && isAlreadyImported(accountId, repoURL, gitEntityInfo.getFilePath())) {
> ```

As discussed, we can avoid another condition check here.

**Comment by mohitg0795** [comment]
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7pE5NH0K4SDF0rL215:bny9du6) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
> ```Java
> @NoArgsConstructor
> @AllArgsConstructor
> public class GitImportInfoDTO {
> ```

This file shouldn't be in 900-git-sync-sdk, please move to other module.

**Comment by mohitg0795** [comment]
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7tACj5DQIlPefT8C4u:b-ycfcd0) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

We have kept it here because that's where the request DTOs for create, get, update etc are. Once again, they all have the same purpose and should be kept together.
</blockquote></details>

I think we discussed this right previously on our call that we will move it to any other module other than 900-git-sdk as this doesn't belong rightly here.
cc @archit-harness

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 15 file(s) to review. 1 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 440 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pChtw9PQZasdYmCBI:-N7tE6Ro11q1YO34GSvd:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L440)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

accountIdentifier*
</blockquote></details>

Done.
___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7tEGxA6JTmz-Rcowhf:b-yn5ztg) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I think we discussed this right previously on our call that we will move it to any other module other than 900-git-sdk as this doesn't belong rightly here.
cc @archit-harness  
</blockquote></details>

This is currently on hold, waiting for the final call.

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 15 file(s) to review. 1 change requests. 10. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pD4oGCynfJs5kBEOJ:-N7tEhnN-yop4a1HwJCO:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

As discussed, we can avoid another condition check here.
</blockquote></details>

Done.
___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 455 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pBTr3CMzc7ZyeztTw:-N7tEiWBATIksfmYoa2Y:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L455)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

this shouldn't be public. If required in UTs, lets make it protected.
</blockquote></details>

Done.
___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 455 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pBfaSC5SYSIXbX1FM:-N7tEj3a03YrI82vnAKr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L455)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

should be accountIdentifier
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7tXoFJ1_LPDOlmXk4p:blkuu08) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

This is currently on hold, waiting for the final call.
</blockquote></details>

For more info: https://harness.slack.com/archives/C0399D8B6AX/p1658821480521669.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 2 of 15 files at r2, 3 of 5 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 4 of 4 files at r6, 3 of 6 files at r7, 1 of 1 files at r8, 2 of 3 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 15 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r8, 2 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 12 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r10](https://reviewable.io/reviews/harness/harness-core/35573#-N7taHX2D9a3ThLQYWUW:-N7taHX30J1IlDOVvcPT:b-hzwuih) ([raw file](https://github.com/harness/harness-core/blob/b8902f1ecae2bd77f6aef75da553b75a6357aba3/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
> ```Java
>           gitEntityInfo.getFilePath());
>     } else if (isAlreadyImported(accountIdentifier, repoURL, gitEntityInfo.getFilePath())) {
>       throw new GitYamlException("The Requested YAML has already been imported.");
> ```

We still have not added repo url and filepath here



_Code quote:_
```Java
throw new GitYamlException("The Requested YAML has already been imported.");
```
___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7taPvCDhzCIh53flw4:b-yp005x) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

For more info: https://harness.slack.com/archives/C0399D8B6AX/p1658821480521669.
</blockquote></details>

@senjucanon2 can you do the needful here

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 12 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 449 at r10](https://reviewable.io/reviews/harness/harness-core/35573#-N7taHX2D9a3ThLQYWUW:-N7tn2fXDb91di-_mTHF:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b8902f1ecae2bd77f6aef75da553b75a6357aba3/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L449)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

We still have not added repo url and filepath here
</blockquote></details>

Done.
___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java` line 31 at r9](https://reviewable.io/reviews/harness/harness-core/35573#-N7pE5NH0K4SDF0rL214:-N7tn5hDADoh9q39EgkA:bn66d0i) ([raw file](https://github.com/harness/harness-core/blob/bf02e16d4ff8ee54f63b9e623db2cf629b86cbe0/900-git-sync-sdk/src/main/java/io/harness/gitsync/interceptor/GitImportInfoDTO.java#L31)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@senjucanon2 can you do the needful here
</blockquote></details>

As discussed with Mohit and Naman, this has been stalled as of now, and will be picked up as task with a new ticket.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 10 of 15 files at r2, 1 of 6 files at r7, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35573)* status: :shipit: complete! ✓

**Pull request merged**